cpp/src/arrow/flight/sql/odbc/flight_sql/config/configuration.cc

```diff
@@ -34,19 +34,22 @@ static const char DEFAULT_USE_CERT_STORE[] = TRUE_STR;
 static const char DEFAULT_DISABLE_CERT_VERIFICATION[] = FALSE_STR;
 
 namespace {
-std::string ReadDsnString(const std::string& dsn, const std::string& key,
+std::string ReadDsnString(const std::string& dsn, const std::string_view& key,
                           const std::string& dflt = "") {
 #define BUFFER_SIZE (1024)
   std::vector<char> buf(BUFFER_SIZE);
 
-  int ret = SQLGetPrivateProfileString(dsn.c_str(), key.c_str(), dflt.c_str(), buf.data(),
-                                       static_cast<int>(buf.size()), "ODBC.INI");
+  std::string key_str = std::string(key);
```

Comment:
You should be able to use key_str.data() (and const-cast it if needed) because your string_views are built on string literals, so it is safe to assume they are NUL-terminated. This will avoid the copy.

Reply:
Good point, switched to use `key.data` and removed the copy